susemanager-utils/susemanager-sls/src/beacons/pkgset.py

```diff
@@ -1,45 +1,38 @@
 # -*- coding: utf-8 -*-
-'''
+"""
 Watch RPM or DPkg database via cookies and fire
 an event to the SUSE Manager if that has been changed.
 
 Author: Bo Maryniuk <bo@suse.de>
-'''
+"""
 
 from __future__ import absolute_import
-from salt.utils.odict import OrderedDict
 import os
-import logging
-log = logging.getLogger(__name__)
 
 
-__virtualname__ = 'pkgset'
+__virtualname__ = "pkgset"
 
-PKG_PLUGINS = OrderedDict([
-        ("/usr/lib/zypp/plugins/commit/venv-zyppnotify", "/var/cache/venv-salt-minion/rpmdb.cookie"),
-        ("/usr/share/yum-plugins/venv-yumnotify.py",     "/var/cache/venv-salt-minion/rpmdb.cookie"),
-        ("/usr/bin/venv-dpkgnotify",                     "/var/cache/venv-salt-minion/dpkg.cookie"),
-        ("/usr/lib/zypp/plugins/commit/zyppnotify",      "/var/cache/salt/minion/rpmdb.cookie"),
-        ("/usr/share/yum-plugins/yumnotify.py",          "/var/cache/salt/minion/rpmdb.cookie"),
-        ("/usr/bin/dpkgnotify",                          "/var/cache/salt/minion/dpkg.cookie")
-    ])
+PKGSET_COOKIES = (
+    "/var/cache/venv-salt-minion/rpmdb.cookie",
+    "/var/cache/venv-salt-minion/dpkg.cookie",
+    "/var/cache/salt/minion/rpmdb.cookie",
+    "/var/cache/salt/minion/dpkg.cookie",
+)
 COOKIE_PATH = None
 
 
 def __virtual__():
-    global COOKIE_PATH
+    return __virtualname__
 
-    for plug, cookie in PKG_PLUGINS.items():
-        if os.path.exists(plug):
-            COOKIE_PATH = cookie
-            break
 
-    return COOKIE_PATH and __virtualname__ or False
+def validate(config):
+    return True, "There is nothing to validate"
```

Comment:
Are you sure that we need this? https://docs.saltproject.io/en/latest/topics/beacons/#writing-beacon-plugins says this function is optional.

Reply:
It's there to reduce the noise in DEBUG logging. Otherwise it prings the debug message that the function is missing in the module.

Reply:
I think the problem is that, even if the function is optional, if it is not present, then salt will spam logs messages every 10 seconds, of course only when `log_level: debug` even so, this is unfortunate as I will contaminate all debug logs with this repeated message.

Reply:
Could you please add a comment in the source code explaining that this is the reason for including this function?

Reply:
Appended with https://github.com/uyuni-project/uyuni/pull/4427/commits/5d168f6c5826f72f39b639bb10166c994d42b965